### PR TITLE
Refactor transportservice request

### PR DESCRIPTION
**Pull request by mloufra**
### Description
Creating a new method `sendGenericRequestWithExceptionHandling` to handle the duplicate code for transport request method `sendClusterStateRequest`, `sendClusterSettingsRequest`, `sendLocalNodeRequest`. 

### Issues Resolved
https://github.com/opensearch-project/opensearch-sdk-java/issues/116

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dbwiddis — APPROVED** [review]
LGTM!

**Review by vibrantvarun — COMMENTED** [review]
Does the methods implemented below, have been tested by either mocking them or passing the dummy objects on the local machine, in order to ensure the right execution of it?

File: src/main/java/org/opensearch/sdk/ExtensionsRunner.java [hunk 1]
```diff
@@ -314,24 +317,32 @@ public void sendRegisterCustomSettingsRequest(TransportService transportService)
         }
     }
 
+    private void sendGenericRequestWithExceptionHandling(
+        TransportService transportService,
+        RequestType requestType,
+        String orchestratorNameString,
+        TransportResponseHandler<? extends TransportResponse> responseHandler
+    ) {
```
[the remark this answers is not in the document]
Comment (mloufra) [reply src/main/java/org/opensearch/sdk/ExtensionsRunner.java:325]: This method is used to handle the duplicate code for `sendClusterStateRequest`, `sendClusterSettingsRequest` and `sendLocalNodeRequest`. When we do grade check, test for these methods passed can prove method `sendGenericRequestWithExceptionHandling` is on the right execution.

[the remark this answers is not in the document]
Comment (vibrantvarun) [reply src/main/java/org/opensearch/sdk/ExtensionsRunner.java:325]: Got it.

**Review by vibrantvarun — APPROVED** [review]

**Pull request merged**